tensorflow/tools/ci_build/builds/pip.sh

```diff
@@ -127,25 +136,43 @@ echo "Installing pip whl file: ${WHL_PATH}"
 
 # Create temporary directory for install test
 VENV_DIR="${PIP_TEST_ROOT}/venv"
-rm -rf "${VENV_DIR}" && mkdir -p "${VENV_DIR}"
-echo "Create directory for virtualenv: ${VENV_DIR}"
+if [[ -d "${VENV_DIR}" ]]; then
+  echo "virtualenv directory already exists: ${VENV_DIR}"
+else
+  echo "Creating virtualenv directory: ${VENV_DIR}"
+  mkdir -p ${VENV_DIR} || \
+      die "FAILED to create virtualenv directory: ${VENV_DIR}"
+fi
 
 # Verify that virtualenv exists
 if [[ -z $(which virtualenv) ]]; then
   die "FAILED: virtualenv not available on path"
 fi
 
-virtualenv -p "${PYTHON_BIN_PATH}" "${VENV_DIR}" ||
-die "FAILED: Unable to create virtualenv"
+virtualenv -p "${PYTHON_BIN_PATH}" "${VENV_DIR}" || \
+    die "FAILED: Unable to create virtualenv"
+
+source "${VENV_DIR}/bin/activate" || \
+    die "FAILED: Unable to activate virtualenv"
 
-source "${VENV_DIR}/bin/activate" ||
-die "FAILED: Unable to activate virtualenv"
 
 # Install the pip file in virtual env
 pip install -v ${WHL_PATH} \
 && echo "Successfully installed pip package ${WHL_PATH}" \
 || die "pip install (without --upgrade) FAILED"
 
+# Install extra pip packages required by the test-on-install
+if [[ ! -z "${TF_BUILD_INSTALL_EXTRA_PIP_PACKAGES}" ]]; then
+  INSTALL_EXTRA_PIP_PACKAGES="${TF_BUILD_INSTALL_EXTRA_PIP_PACKAGES}"
```

Comment:
Why not to do this as the other things... at top use TF_BUILD_INSTALL_EXTRA_PIP_PACKAGES=${TF_BUILD_INSTALL_EXTRA_PIP_PACKAGES:-scipy sklearn}"

It makes it clear at top what arguments it takes, what are defaults and makes the code easier to follow if things are immutable (in any language).